Apple Silicon reenable 49110 failures

sdmaclea opened the pull request:
These issues were likely fixed in CI by dotnet/core-eng#12776

Fixes #49110

sdmaclea committed:
Apple Silicon reenable 49110 failures

safern commented on src/libraries/tests.proj:
```diff
@@ -299,56 +299,6 @@
     <!-- ActiveIssue Apple Silicon No usable version of libssl was found https://github.com/dotnet/runtime/issues/49083 -->
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Security.Cryptography.Algorithms/tests/System.Security.Cryptography.Algorithms.Tests.csproj" />
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Security.Cryptography.OpenSsl/tests/System.Security.Cryptography.OpenSsl.Tests.csproj" />
-    <!-- ActiveIssue XUnit crashes on Apple Silicon https://github.com/dotnet/runtime/issues/49110 -->
```
Could all these that were crashing uncover other test failures? Let's make sure before merging that the build is still green.

sdmaclea commented:
/azp run runtime-libraries-coreclr outerloop-osx

sdmaclea commented:
Manually triggered https://dev.azure.com/dnceng/public/_build/results?buildId=1119668&view=results

ericstj commented:
@sdmaclea does this need a rerun or do you think these weren't fixed by  dotnet/core-eng#12776?

sdmaclea commented:
These weren't fixed by dotnet/core-eng#12776.  

Unfortunately most of these issues do not reproduce (easily?) locally anymore.  

The original runs didn't generate core dumps.  The coredumps have since been fixed.  I planned to rebase and rerun these to see if we could get some more info.

/cc @janvorli @mangod9 @sandreenko

sdmaclea commented:
Rebased and manually triggered https://dev.azure.com/dnceng/public/_build/results?buildId=1176673&view=results

sdmaclea commented:
/azp run runtime-libraries-coreclr outerloop-osx

sdmaclea commented:
There are some coredumps for a bus error and a segmentation fault https://dev.azure.com/dnceng/public/_build/results?buildId=1176673

There is also this interesting delegate test failure
```console
  Starting:    System.Runtime.Tests (parallel test collections = on, max threads = 8)
    System.Tests.DelegateTests.DynamicInvoke_DefaultParameter_AllPrimitiveParametersWithSomeExplicitValues [FAIL]
      Assert.Equal() Failure
                                       â†“ (pos 38)
      Expected: Â·Â·Â· -101, -3, 104, -5, 106, -7, 108, 9.1, 111.12
      Actual:   Â·Â·Â· -101, -3, 104, -5, 4294967291, -7, 108, 9.1, 111.12
                                       â†‘ (pos 38)
      Stack Trace:
        /_/src/libraries/System.Runtime/tests/System/DelegateTests.cs(253,0): at System.Tests.DelegateTests.DynamicInvoke_DefaultParameter_AllPrimitiveParametersWithSomeExplicitValues()
    System.Tests.DelegateTests.DynamicInvoke_DefaultParameter_AllPrimitiveParametersWithMissingValues [FAIL]
      Assert.Equal() Failure
                                       â†“ (pos 33)
      Expected: Â·Â·Â·, 2, -1, -3, 4, -5, 6, -7, 8, 9.1, 11.12
      Actual:   Â·Â·Â·, 2, -1, -3, 4, -5, 4294967291, -7, 8, 9.1, 11.12
                                       â†‘ (pos 33)
    System.Tests.DelegateTests.DynamicInvoke_DefaultParameter_AllPrimitiveParametersWithAllExplicitValues [FAIL]
      Assert.Equal() Failure
                                       â†“ (pos 45)
      Expected: Â·Â·Â·1, -103, 104, -105, 106, -107, 108, 109.1, 111.12
      Actual:   Â·Â·Â·1, -103, 104, -105, 4294967191, -107, 108, 109.1, 111.12
                                       â†‘ (pos 45)
      Stack Trace:
        /_/src/libraries/System.Runtime/tests/System/DelegateTests.cs(228,0): at System.Tests.DelegateTests.DynamicInvoke_DefaultParameter_AllPrimitiveParametersWithAllExplicitValues()
```

Looks like an ABI issue which was covered by the disabled tests.  Perhaps it was fixed by #53250 which was merged after this was triggered.

sdmaclea commented:
This run the same test suites failed. https://dev.azure.com/dnceng/public/_build/results?buildId=1177129

But the interesting case is gone.  Perhaps it is fixed or intermittent.

sdmaclea committed:
Restore test disables

System.Numerics.Vectors & System.Runtime both failed
for both runs.  Restore disables of these tests.

sdmaclea commented:
/azp run runtime-libraries-coreclr outerloop-osx
